app/code/Magento/Catalog/etc/db_schema.xml

```diff
@@ -573,6 +577,27 @@
             <column name="store_id"/>
         </index>
     </table>
+    <table name="catalog_compare_list" resource="default" engine="innodb" comment="Catalog Compare List with hash Table">
+        <column xsi:type="int" name="list_id" padding="10" unsigned="true" nullable="false"
+                identity="true" comment="Compare List ID"/>
+        <column xsi:type="int" name="visitor_id" unsigned="true" nullable="false" identity="false"
+                default="0" comment="Visitor ID"/>
+        <column xsi:type="int" name="customer_id" padding="10" unsigned="true" nullable="true" identity="false"
+                comment="Customer ID"/>
+        <constraint xsi:type="primary" referenceId="PRIMARY">
+            <column name="list_id"/>
+        </constraint>
+        <constraint xsi:type="foreign" referenceId="CATALOG_COMPARE_LIST_CUSTOMER_ID_CUSTOMER_ENTITY_ENTITY_ID"
+                    table="catalog_compare_list" column="customer_id" referenceTable="customer_entity"
+                    referenceColumn="entity_id" onDelete="CASCADE"/>
+        <index referenceId="CATALOG_COMPARE_LIST_LIST_ID" indexType="btree">
+            <column name="list_id"/>
+        </index>
+        <index referenceId="CATALOG_COMPARE_LIST_VISITOR_ID_CUSTOMER_ID" indexType="btree">
+            <column name="visitor_id"/>
```

Comment:
@Usik2203 Why do we need the visitor_id? How is it being generated for a GraphQl client?

Reply:
Hi @prabhuram93 
`visitor_id` it's value come from table `customer_visitor` field `visitor_id` 
It need for detecting what compare list should remove or add to compare list for guest users

Scenario: https://drive.google.com/file/d/1jfEdoXa4tArU7z1rhSHgZX08Gxr4DzQL/view?usp=sharing

Reply:
**UPD**
@prabhuram93 
I have got rid from `visitor_id` field in` DB`  as redundant after applying changes based on new schema
Thanks